src/ucp/core/ucp_ep.c

```diff
@@ -320,6 +328,9 @@ ucs_status_t ucp_worker_create_ep(ucp_worker_h worker, unsigned ep_init_flags,
     return UCS_OK;
 
 err_destroy_ep_base:
+    if (ucp_ep_ext_control(ep)->local_ep_id != UCP_EP_ID_INVALID) {
+        ucp_ep_set_failed(ep);
```

Comment:
why need to set UCP_EP_FLAG_FAILED flag here?

Reply:
need to avoid releasing id (which was not allocated yet) in this constructor error flow, by moving release id to a better place

Reply:
it is not needed at all